### PR TITLE
[Staging] Freeform Topics indexing

### PR DESCRIPTION
**Pull request by phillipahereza**
## Description

- Update topics indexing to work with the new **freeform** structure,
- Add API endpoint for everywhere search `/api/search/q=fgm`, and
- Add summary field to indicators.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: takwimu/management/commands/update_topics_index.py [hunk 1]
```diff
@@ -25,45 +34,127 @@ def index_topics(self, queryset, search_backend):
...
-                                data['body'],
-                                data['metadata'],
-                                parent_page_id,
-                                parent_page_type
-                            )
-                            self.stdout.write(
-                                search_backend.es_index + ": Indexing widget '%s result %s'" % (
+                topic_body = topic['value'].get('body', '')
+                for k in topic_body:
+                    if k['type'] == 'text':
```
Comment (kilemensi) [inline takwimu/management/commands/update_topics_index.py]:
```
Traceback (most recent call last):
  File "manage.py", line 22, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.7/site-packages/django/core/management/__init__.py", line 356, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/src/takwimu/takwimu/management/commands/update_topics_index.py", line 25, in handle
    self.index_topics(ProfileSectionPage.objects.live(), topic_search)
  File "/src/takwimu/takwimu/management/commands/update_topics_index.py", line 40, in index_topics
    if k['type'] == 'text':
TypeError: string indices must be integers
Makefile:17: recipe for target 'update_index' failed
make: *** [update_index] Error 1
```

@phillipahereza

Comment (kilemensi) [reply takwimu/management/commands/update_topics_index.py]: Could be the **old** topic block though... Let me rebuild and try again.

Comment (phillipahereza) [reply takwimu/management/commands/update_topics_index.py]: @kilemensi probably this error `TypeError: string indices must be integers` is from that old topic block

Comment (kilemensi) [reply takwimu/management/commands/update_topics_index.py]: Works @phillipahereza

File: takwimu/management/commands/update_topics_index.py [hunk 2]
```diff
@@ -25,45 +34,127 @@ def index_topics(self, queryset, search_backend):
...
+        options.add_argument('disable-gpu')
+        browser = webdriver.Chrome(options=options)
+
+        server_url = os.getenv('HURUMAP_URL', 'http://localhost:8000')
+
+        if server_url.endswith('/'):
+            # remove trailing slash
+            server_url = server_url[:-1]
+
+        urls = {'Kenya': 'profiles/country-KE-kenya/',
```
Comment (kilemensi) [inline takwimu/management/commands/update_topics_index.py]: Use countries list, they have `iso_code` & `slug`

File: takwimu/management/commands/update_topics_index.py [hunk 3]
```diff
@@ -25,45 +34,127 @@ def index_topics(self, queryset, search_backend):
...
+
+    def index_hurumap(self, search_backend):
+        options = webdriver.ChromeOptions()
+        options.add_argument('headless')
+        # chrome can not be run as root, which is the case in docker
+        options.add_argument('no-sandbox')
+        options.add_argument('disable-gpu')
+        browser = webdriver.Chrome(options=options)
+
+        server_url = os.getenv('HURUMAP_URL', 'http://localhost:8000')
```
Comment (kilemensi) [inline takwimu/management/commands/update_topics_index.py]: use `settings.HURUMAP_URL`, it should already load default url.

File: takwimu/management/commands/update_topics_index.py [hunk 4]
```diff
@@ -25,45 +34,127 @@ def index_topics(self, queryset, search_backend):
...
+                    # self.stdout.write(str(viz))
+                    labels = ' '.join(
+                        [i.text for i in viz.select('span.x.axis.label')])
+
+                    qualifiers = ' '.join(
+                        [i.text for i in
+                         viz.find_all('span', class_=['chart-qualifier'])])
+
+                    id = f"{country}-{viz['id']}"
+                    link = f"{server_url}/{url}#{viz['id']}"
```
Comment (kilemensi) [inline takwimu/management/commands/update_topics_index.py]: Use relative urls for internal links... full urls could end up being opened in new tab in the front end (depending on component used).

File: takwimu/urls.py [hunk 5]
```diff
@@ -34,6 +35,7 @@
     url(r'^sdgs/$', SDGIndicatorView.as_view(), name='sdgs'),
     url(r'^feed/$', CountryProfileFeed(), name='rss_feed'),
     url(r'^search/$', SearchView.as_view(), name='search'),
+    url(r'^api/search/$', SearchAPIView.as_view(), name='api-search'),
```
Comment (kilemensi) [inline takwimu/urls.py]: Just a comment, this doesn't affect Wagtail/Wazimap API at all, right?

Comment (phillipahereza) [reply takwimu/urls.py]: @kilemensi Wagtail API is at `api/v2`

**Comment by phillipahereza** [comment]
> Looks good @phillipahereza.
> 
> Why are all summaries pretty much empty/null though?

@kilemensi I had some brilliant idea for summaries but now I can't remember it, I've removed that field till I can remember what data they were supposed to hold

File: takwimu/models/dashboard.py [hunk 6]
```diff
@@ -340,6 +340,8 @@ class Meta:
 class IndicatorBlock(blocks.StructBlock):
     title = blocks.CharBlock()
     widget = IndicatorWidgetsBlock(min_num=1, max_num=1)
+    summary = blocks.RichTextBlock(required=False,
+                                   default='<p>Lorem ipsum dolor sit amet, adipiscing elitauris con lorem ipsum dolor sit amet.</p>')
```
Comment (karimkawambwa) [inline takwimu/models/dashboard.py]: Maybe instead of `lorem ipsum`, default should be empty to be safe.

Comment (kilemensi) [reply takwimu/models/dashboard.py]: Makes sense.

**Pull request merged**